features/329-identity-apis/design.md

```diff
@@ -132,3 +132,11 @@ Conceptually, identities are immutable. Developer should instead:
 1. Create new identity
 2. Ensure the new identity is in desired state (e.g. verification)
 3. Delete old identity
+
+## Adding login ID without existing login IDs
+If user without existing login IDs tried to add a new login ID (e.g. a SSO
+user), a password identity would be created, but the password would not be
+known. In this case, the password identity would be created without password
+(i.e. user cannot login using this identity). User should change the password
+using re-authentication through SSO to enable logging in with the password
+identity. Note that reauthentication must be enabled for this to work.
```

Comment:
Maybe we should also mention the scenario that `Misconfiguration to required login ID amount may cause inability to add login ID`?

refs: https://github.com/SkygearIO/skygear-server/pull/1178#discussion_r362147743

Reply:
After some thoughts, the minimum number of login ID is actually quite hard to uphold. For example, it is completely fine for the user to sign up as a new user via SSO. In this case, the user does not have any login IDs. If the developer sets both the minimum number of `email` and `phone` to be 1, the constraint cannot be upheld in `addLoginID` because only one login ID can be added at a time. It turns out that the constraint is only checked during `signup`. Maybe we should remove the minimum and let user to check this in `before_user_create` hook?

Reply:
@chpapa what do you think about removing the configuration of minimum amount of login ID?

Reply:
I don't see anything particular problem with it.